components/operations-controller/main.go

```diff
@@ -52,33 +65,54 @@ func init() {
 }
 
 func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
-	flag.Parse()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	defer cancel()
+
+	term := make(chan os.Signal)
+	signal.HandleInterrupts(ctx, cancel, term)
+
+	env, err := env.Default(ctx, config.AddPFlags)
+	fatalOnError(err)
+
+	cfg, err := config.New(env)
+	fatalOnError(err)
+
+	err = cfg.Validate()
+	fatalOnError(err)
 
 	ctrl.SetLogger(zap.New(zap.UseDevMode(devLogging)))
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:             scheme,
-		MetricsBindAddress: metricsAddr,
+		MetricsBindAddress: cfg.Server.MetricAddress,
 		Port:               port,
-		LeaderElection:     enableLeaderElection,
+		LeaderElection:     cfg.Server.EnableLeaderElection,
 		LeaderElectionID:   leaderElectionID,
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	if err = (&controllers.OperationReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("Operation"),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
+	httpClient := http.Client{
+		Transport: httputil.NewHTTPTransport(cfg.HttpClient),
```

Comment:
The transport can be wrapped as in the gql client:
```
httputil.NewCorrelationIDTransport(httputil.NewErrorHandlerTransport(httputil.NewHTTPTransport(httpCfg)))
```

Reply:
I agree with the `ErrorHandlerTransport`, but not sure we want to wrap the default http client with the `CorrelationIDTransport` decorator. Currently the default http client is used for webhook requests and the correlation ID which was used with the original async mutation is reused (if provided, in our current use case with it will be). Maybe it would be confusing to send two correlation ids to them.

The default http client is also used for making the Update Operation status requests to the Director from Controller side. There again I'm not sure if we should generate random correlation ids rather than reusing the one from the original mutation.

Reply:
Correlation IDs attaching has been added. I wasn't able to find anything around the `controller-runtime` regarding extracting additional tracing info. Maybe we can re-visit this in a separate tast.